File: nittyGritty/src/watch/my/beans/Account.java
package watch.my.beans;

public class Account 
{

	private String accountNumber;
    private String accountHolderName;
    private double accountBalance;
    
    public Account( String accountHolderName, String accoutNumber)
    {
    	this.accountNumber 		= accoutNumber;
    	this.accountHolderName	= accountHolderName;
    	this.accountBalance		= 0;
    	
    	
    	
    }

    public static void transfer(Account from, Account to, double amount)
	{
			from.accountBalance 		= from.accountBalance - amount;
			to.accountBalance 		= to.accountBalance + amount;
		
	}
    
    public void updateBalance(double amount)
	{
			this.accountBalance = this.accountBalance + amount;
		
	}
    
    public double getBalance()
	{
			return this.accountBalance;
		
	}
    
    public String getHolderNAme()
    {
    			return this.accountHolderName;
    }
    


}

File: nittyGritty/src/watch/my/beans/scale/Account.scala
package watch.my.beans.scale

class Account (accountHolderNameIn: String, accoutNumberIn: String) 
{
    
    var accountHolderName        = accountHolderNameIn
    var accoutNumber             = accoutNumberIn
    var accountBalance: Double   = 0
    
    def updateBalance(amount: Double) 
       {
      accountBalance = accountBalance + amount 
       }
    
    
   


}

object Account
{
    def transfer(from: Account,to: Account, amount: Double) 
       {
      from.accountBalance = from.accountBalance - amount
      to.accountBalance = to.accountBalance + amount
       }
  
}